CHANGELOG.md

```diff
@@ -1,5 +1,46 @@
 # Changelog
 
+## [v0.12.0](https://github.com/delta-io/delta-kernel-rs/tree/v0.12.0/) (2025-06-04)
+
+[Full Changelog](https://github.com/delta-io/delta-kernel-rs/compare/v0.11.0...v0.12.0)
+
+### 🏗️ Breaking changes
+1. Remove `GlobalScanState`: instead use new `Scan` APIs directly (`logical_schema`, `physical_schema`, etc.) ([#947])
+1. table feature enums are now `internal_api` (not public, unless `internal-api` flag is set) ([#998])
+
+### 🚀 Features / new APIs
+
+1. Use compacted log files in log-replay ([#950])
+2. New `#[derive(IntoEngineData)]` proc macro ([#830])
+3. Add support for kernel default expression evaluation ([#979])
+4. Add `debug_assert` to the constructor of `struct ListedLogFiles` ([#986])
```

Comment:
Can we move this to "Breaking changes", but note **in bold** that it's _only_ breaking if you compile debug? Or at least call it out a bit more.

Reply:
yea sg - not sure if it qualifies as a breaking change (semantics are same, just that we panic if you break them in debug mode?)

maybe just add a flag to the line here then?

Reply:
Yeah, flag on the line is fine